client/src/bastionlab/polars/remote_polars.py

```diff
@@ -313,64 +315,161 @@ def join_asof(
+                .filter(pl.col(col_y) != None)
+                .select([pl.col(col_y), pl.col(col_x)])
+                .apply_udf([col_x], model)
+                .groupby([pl.col(col_x), pl.col(col_y)])
+                .agg(pl.count())
+                .sort(pl.col(col_x))
+                .collect()
+                .fetch()
+                .to_pandas()
+            )
+            my_cmap = sns.color_palette("Blues", as_cmap=True)
+            pivot = df.pivot(index=col_y, columns=col_x, values="count")
+            if "cmap" not in kwargs:
+                kwargs["cmap"] = my_cmap
+            ax = sns.heatmap(pivot, **kwargs)
+            ax.invert_yaxis()
 
     def curveplot(
         self: LDF,
-        col_x: str,
-        col_y: str,
-        bins: int = 10,
+        x: str,
+        y: str,
         order: int = 3,
         ci: Union[int, None] = None,
         scatter: bool = False,
         **kwargs,
     ):
-        model = ApplyBins(bins)
-        df = (
-            self.filter(pl.col(col_x) != None)
-            .filter(pl.col(col_y) != None)
-            .select([pl.col(col_y), pl.col(col_x)])
-            .apply_udf([col_x], model)
-            .groupby(pl.col(col_x))
-            .agg(pl.col(col_y).count())
-            .sort(col_x)
-            .collect()
-            .fetch()
-            .to_pandas()
-        )
-        sns.regplot(
-            x=df[col_x], y=df[col_y], order=order, ci=ci, scatter=scatter, **kwargs
-        )
-
-    def scatterplot(self: LDF, col_x: str, col_y: str, bins=5, **kwargs):
-        model = ApplyBins(bins)
-        df = (
-            self.filter(pl.col(col_x) != None)
-            .filter(pl.col(col_y) != None)
-            .select([pl.col(col_y), pl.col(col_x)])
-            .apply_udf([col_x], model)
-            .groupby(pl.col(col_x))
-            .agg(pl.col(col_y).count())
-            .sort(col_x)
-            .collect()
-            .fetch()
-            .to_pandas()
-        )
-        sns.scatterplot(x=df[col_x], y=df[col_y], **kwargs)
+        """Plot data and a linear regression model fit.
+        Curveplot filters data down to necessary columns only and then calls Seaborn's regplot function.
+        Args:
+            x (str): The name of column to be used for x axes.
+            y (str): The name of column to be used for y axes.
+            order (int): If order is greater than 1, Seaborn's regplot uses numpy.polyfit to estimate a polynomial regression. Default value is 3.
+            ci: Size of the confidence interval for the regression estimate. Default value is None.
+            scatter (bool): Option to display a scatterplot of data with the underlying observations. Default value is False.
+            **kwargs: Other keyword arguments that will be passed to Seaborn's regplot function.
+        """
+        for col in [x, y]:
+            if not col in self.columns:
+                print("Error: ", col, " does not exist in dataframe")
```

Comment:
Also, like on the others error print, I think an exception is more suited to show an error.